kernel/sched.c

```diff
@@ -368,7 +368,7 @@ static void pend(struct k_thread *thread, _wait_q_t *wait_q, s32_t timeout)
 	}
 
 	if (timeout != K_FOREVER) {
-		s32_t ticks = _TICK_ALIGN + z_ms_to_ticks(timeout);
+		s32_t ticks = _TICK_ALIGN + k_ms_to_ticks_ceil32(timeout);
```

Comment:
what about something like: `s32_t ticks =timeout > 0 ? (_TICK_ALIGN + k_ms_to_ticks_ceil32(timeout)) : 0;`

Reply:
Likewise, this is just an automated change to code we've been carrying already.